### ISSUE TITLE
Neural Machine Translation via Binary Code Prediction

### ISSUE THREAD

**Issue by ymym3412** [ymym3412/acl-papers#76]
## 0. 論文
[Neural Machine Translation via Binary Code Prediction]
Yusuke Oda; Philip Arthur; Graham Neubig; Koichiro Yoshino; Satoshi Nakamura

## 1. どんなもの？
出力を単語の2進数表現で行うニューラル機械翻訳モデル

## 2. 先行研究と比べてどこがすごい？
メモリ効率及び計算効率の両方を重視している

## 3. 技術や手法のキモはどこ？
Softmaxの次元数は本来Vocab数だが、出力を2進数表現にすれば必要な次元数はlog2(V)に抑えられメモリ効率と計算効率を向上できる。
ただ2進数表現をそのまま予測すると翻訳の性能に影響を及ぼすため「Softmaxと2進数表現のモデルを組み合わせる(Hybrid)」「ビットに冗長性を持たせる(EC)」ことでメモリ効率と計算効率を向上させつつ翻訳性能の劣化を抑えている。

![image](https://user-images.githubusercontent.com/9605058/33269095-d700f7f6-d3c2-11e7-8c05-3c5219d77008.png)

![image](https://user-images.githubusercontent.com/9605058/33269113-e5f07944-d3c2-11e7-82fe-28ff3937cd5c.png)

## 4. どうやって有効だと検証した？
ASPECコーパスとBTECコーパスで通常のSoftmaxのモデルとBLEUスコアで比較した。
Hybrid+ECのモデルがBTECコーパスではSoftmaxより高性能、ASPECコーパスでは通常のSoftmaxのモデルに迫るスコアを記録した。

## 5. 議論はある？
BTECコーパスで2進表現モデルが勝ったのはコーパスの複雑度によるものとのこと

## 6. 次に読むべき論文は？
計算効率を上げる手法
hierarchical softmax
[Hierarchical probabilistic neural network language model](https://www.iro.umontreal.ca/~lisa/pointeurs/hierarchical-nnlm-aistats05.pdf)
Frederic Morin and Yoshua Bengio

differential softmax
[Strategies for training large vocabulary neural language models.](https://arxiv.org/abs/1512.04906)
Wenlin Chen, David Grangier, and Michael Auli
